models/DataObject/AbstractObject.php

```diff
@@ -406,9 +406,9 @@ public function getSiblings(
                 if ($this->getId()) {
                     $list->addConditionParam('id != ?', $this->getId());
                 }
-                $list->setOrderKey('key');
                 $list->setObjectTypes($objectTypes);
-                $list->setOrder('asc');
+                $list->setOrderKey($this->getChildrenSortBy());
```

Comment:
I think, we need the settings from the parent and not from the own object.

Reply:
Indeed. However question remains if the is the desired behavior.
And the question is getting more important since we now also have an additional `getParent()` call which potentially slows down execution.

Reply:
Maybe change `if ($this->getParentId()) {` to `if ($parent = $this->getParent()) {` and then use that `$parent` object in the updated rows to only have one call?